core/domain/stats_jobs_one_off_test.py

```diff
@@ -28,11 +34,663 @@
+            event_schema_version=2).put()
+        stats_models.StateCompleteEventLogEntryModel(
+            id='id2',
+            exp_id=self.exp_id,
+            exp_version=1,
+            state_name=self.state_a,
+            session_id='session_id_3',
+            time_spent_in_state_secs=1.0,
+            event_schema_version=1).put()
+        stats_models.StateCompleteEventLogEntryModel(
+            id='id3',
+            exp_id=self.exp_id,
+            exp_version=2,
+            state_name=self.state_a,
+            session_id='session_id_4',
+            time_spent_in_state_secs=1.0,
+            event_schema_version=2).put()
+        stats_models.StateCompleteEventLogEntryModel(
+            id='id4',
+            exp_id=self.exp_id,
+            exp_version=3,
+            state_name=self.state_b,
+            session_id='session_id_4',
+            time_spent_in_state_secs=1.0,
+            event_schema_version=2).put()
+
+        state_stats_dict = {
+            'total_answers_count_v1': 3,
+            'total_answers_count_v2': 0,
+            'useful_feedback_count_v1': 3,
+            'useful_feedback_count_v2': 0,
+            'total_hit_count_v1': 3,
+            'total_hit_count_v2': 0,
+            'first_hit_count_v1': 3,
+            'first_hit_count_v2': 0,
+            'num_times_solution_viewed_v2': 0,
+            'num_completions_v1': 3,
+            'num_completions_v2': 9
+        }
+        stats_models.ExplorationStatsModel.create(
+            self.exp_id, 1, 0, 0, 0, 0, 0, 0,
+            {
+                self.state_a: state_stats_dict
+            })
+        stats_models.ExplorationStatsModel.create(
+            self.exp_id, 2, 0, 0, 0, 0, 0, 0,
+            {
+                self.state_a: state_stats_dict
+            })
+        stats_models.ExplorationStatsModel.create(
+            self.exp_id, 3, 0, 0, 0, 0, 0, 0,
+            {
+                self.state_a: state_stats_dict
+            })
+
+    def test_standard_operation(self):
+        def mock(x, y, z): #pylint: disable=unused-argument,invalid-name
+            return [{'commit_cmds':[]}, {'commit_cmds':[]}, {
+                'commit_cmds': [{'cmd': exp_domain.CMD_RENAME_STATE,
+                                 'old_state_name': self.state_a,
+                                 'new_state_name': self.state_b}]}]
+        with self.swap(
+            base_models.VersionedModel, 'get_snapshots_metadata',
+            types.MethodType(mock, base_models.VersionedModel)):
+            job_id = (
+                stats_jobs_one_off.RecomputeStatistics.create_new())
+            stats_jobs_one_off.RecomputeStatistics.enqueue(job_id)
+
+            self.assertEqual(
+                self.count_jobs_in_taskqueue(
+                    taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+            self.process_and_flush_pending_tasks()
+
+        model_id = stats_models.ExplorationStatsModel.get_entity_id(
+            self.exp_id, '1')
```

Comment:
Here and below -- not sure why you're passing in strings for the version number, I think it expects int (according to the documentation).

Reply:
Done.